src/native/eventpipe/ep-buffer-manager.c

```diff
@@ -1373,7 +1371,29 @@ ep_buffer_manager_get_next_event (EventPipeBufferManager *buffer_manager)
 	// buffer at the same time.
 	ep_timestamp_t stop_timestamp = ep_perf_timestamp_get ();
 	buffer_manager_move_next_event_any_thread (buffer_manager, stop_timestamp);
+
```

Comment:
Since this is a hot path then maybe you should add logic to only do this when necessary. You detect when there is empty buffers as part of buffer_manager_move_next_event_any_thread so it could have an out parameter indicating if it saw exhausted session states while moving to next event and that could then trigger the cleanup code below.